src/python/pants/bin/local_pants_runner.py

```diff
@@ -325,23 +327,29 @@ def _run(self):
     callbacks = Subsystem.get_streaming_workunit_callbacks(streaming_handlers)
     streaming_reporter = StreamingWorkunitHandler(self._scheduler_session, callbacks=callbacks)
 
+    help_output = self._maybe_handle_help()
+    if help_output is not None:
+      self._exiter.exit(help_output)
+
     with streaming_reporter.session():
       try:
-        self._maybe_handle_help()
         engine_result = self._maybe_run_v2()
         goal_runner_result = self._maybe_run_v1()
       finally:
-        try:
-          self._update_stats()
-          run_tracker_result = self._run_tracker.end()
-        except ValueError as e:
-          # Calling .end() sometimes writes to a closed file, so we return a dummy result here.
-          logger.exception(e)
-          run_tracker_result = PANTS_SUCCEEDED_EXIT_CODE
-
+        run_tracker_result = self._finish_run()
     final_exit_code = self._compute_final_exit_code(
       engine_result,
       goal_runner_result,
       run_tracker_result
     )
     self._exiter.exit(final_exit_code)
+
+  def _finish_run(self):
+    try:
+      self._update_stats()
+      run_tracker_result = self._run_tracker.end()
```

Comment:
for simplicity we should just do `return self._run_tracker.end()`
and `return PANTS_SUCCEEDED_EXIT_CODE` (when we catch an error).
it is less code and more pythonic.